packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -1,35 +1,43 @@
-import { createElement, ReactElement, useReducer, useRef } from "react";
+import { createElement, Dispatch, ReactElement, useCallback, useReducer, useRef } from "react";
 
-import AccordionGroup, { AccGroup } from "./AccordionGroup";
-import { getAccordionGroupsReducer } from "../utils/AccordionGroupStateReducer";
+import AccordionGroup, { AccordionGroupProps } from "./AccordionGroup";
+import {
+    CollapsedAccordionGroupsReducerAction,
+    getCollapsedAccordionGroupsReducer
+} from "../utils/CollapsedAccordionGroupsReducer";
 import { AccordionContainerProps } from "../../typings/AccordionProps";
 import classNames from "classnames";
 
+export type AccordionGroups = Array<Pick<AccordionGroupProps, "header" | "content" | "visible" | "dynamicClassName">>;
+
 export interface AccordionProps extends Pick<AccordionContainerProps, "class" | "style" | "tabIndex"> {
     id: string;
-    groups: AccGroup[];
+    groups: AccordionGroups;
     collapsible: boolean;
+    animateCollapsing?: boolean;
     singleExpandedGroup?: boolean;
+    generateHeaderIcon?: (collapsed: boolean) => ReactElement;
+    showGroupHeaderIcon?: "right" | "left" | "no";
 }
 
 export default function Accordion(props: AccordionProps): ReactElement | null {
-    const previousGroupsPropValue = useRef(props.groups);
+    const reducer = useRef(getCollapsedAccordionGroupsReducer(props.singleExpandedGroup ? "single" : "multiple")); // the accordion group reducer function doesn't need to change during the lifetime of this component, since the singleExpandedGroup won't change.
```

Comment:
Why do we need to pass a ref to the reducer?

Reply:
The `getCollapsedAccordionGroupsReducer` will be called on every rerender when called inside the useReducer function. Making use of a `useCallback` hook doesn't work either, because the `singleExpandedGroup` prop get loads of updates due to the new objects that are being created in the root component.